unit-tests/run-unit-tests.py

```diff
@@ -264,79 +268,127 @@ def check_log_for_fails(log, testname, exe):
+    @abstractmethod
+    def run_test(self):
+        pass
 
-    progress( testname, '>', log, '...' )
-    n_tests += 1
-    try:
-        run( [exe], stdout=log )
-    except FileNotFoundError:
-        error( red + testname + reset + ': executable not found! (' + exe + ')' )
-        continue
-    except subprocess.CalledProcessError as cpe:
-        if not check_log_for_fails(log, testname, exe):
-            # An unexpected error occurred
-            error( red + testname + reset + ': exited with non-zero value! (' + str(cpe.returncode) + ')' )
-
-# If we run python tests with no .pyd/.so file they will crash. Therefore we only run them if such a file was found
-if pyrs:
-    # unit-test scripts are in the same directory as this script
-    for py_test in find(current_dir, '(^|/)test-.*\.py'):
-    
-        testdir = py_test[:-3]                         # "log/internal/test-all"  <-  "log/internal/test-all.py"
-        testparent = os.path.dirname(testdir)          # same as for cpp files
-        if testparent:
-            testname = 'test-' + testparent.replace( '/', '-' ) + '-' + os.path.basename(testdir)[5:]
-        else:
-            testname = testdir
-    
-        log = logdir + '/' + testname + '.log'
-        
-        progress( testname, '>', log, '...' )
-        n_tests += 1
-        test_path = current_dir + os.sep + py_test
-        if linux:
-            cmd = ["python3", test_path]
-        else:
-            cmd = ["py","-3", test_path]
+class PyTest(Test):
+    """
+    Class for python tests
+    """
+    def __init__(self, testname, path_to_test):
+        """
+        :param testname: name of the test
+        :param path_to_test: the relative path from the current directory to the path
+        """
+        global current_dir
+        Test.__init__(self, testname)
+        self.path_to_script = current_dir + path_to_test
+
+    @property
+    def command(self):
+        return [sys.executable, self.path_to_script]
+
+    def run_test(self):
+        log = logdir + os.sep + self.testname + ".log"
+        progress(self.testname, '>', log, '...')
         try:
-            run( cmd, stdout=log )
+            subprocess_run(self.command, stdout=log)
         except FileNotFoundError:
-            error( red + testname + reset + ': file not found! (' + test_path + ')' )
-            continue
+            error(red + self.testname + reset + ': executable not found! (' + self.path_to_script + ')')
         except subprocess.CalledProcessError as cpe:
-            if not check_log_for_fails(log, testname, py_test):
+            if not check_log_for_fails(log, self.testname, self.path_to_script):
                 # An unexpected error occurred
-                cat(log)
-                error(red + testname + reset + ': exited with non-zero value! (' + str(cpe.returncode) + ')')
+                error(red + self.testname + reset + ': exited with non-zero value! (' + str(cpe.returncode) + ')')
+
```

Comment:
Because this function uses elements specific for tests such as testname and logging I thought it wasn't logical to make it a global function

Reply:
What about n_tests?

Reply:
added